python_src/src/lamp_py/migrations/versions/performance_manager/7f4bbec1b669_static_trips_branch_and_trunk_route_id_.py

```diff
@@ -0,0 +1,124 @@
+"""static_trips branch and trunk route_id trigger
+
+Revision ID: 7f4bbec1b669
+Revises: 43153d536c2a
+Create Date: 2023-04-25 10:29:30.479958
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "7f4bbec1b669"
+down_revision = "43153d536c2a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.add_column(
+        "static_trips",
+        sa.Column("branch_route_id", sa.String(60), nullable=True),
+    )
+    op.add_column(
+        "static_trips",
+        sa.Column("trunk_route_id", sa.String(60), nullable=True),
+    )
+    create_red_braintree_branch = """
+        CREATE OR REPLACE FUNCTION red_is_braintree_branch(p_trip_id varchar, p_fk_ts int) RETURNS boolean AS $$ 
+        DECLARE
+            is_braintree boolean;
+        BEGIN 
+            SELECT
+                count(*) > 0
+            INTO
+                is_braintree
+            FROM 
+                static_stop_times sst
+            WHERE 
+                sst.trip_id = p_trip_id
+                AND sst."timestamp" = p_fk_ts
+                AND sst.stop_id IN ('70097', '70098', '70099', '70100', '70101', '70102', '70103', '70104', '70105')
+            ;
+ 
+            RETURN is_braintree;
+        END;
```

Comment:
This function abstracts out some of the business logic required in the `insert_static_trips_branch_trunk` function. 

Will return a true of false, depending on if any of the `stop_id`s for stations after JFK, going to braintree, exist in the trip.